pkg/awsutils/awsutils.go

```diff
@@ -224,8 +225,66 @@ func prometheusRegister() {
 	}
 }
 
+//StringSet is a set of strings
+type StringSet struct {
+	sync.RWMutex
+	data 	sets.String
+}
+
+func (ss *StringSet) AWSStrings() []*string {
+	ss.RLock()
+	defer ss.RUnlock()
+	var dataSlice []*string
+	for key, _ := range ss.data {
+		dataSlice = append(dataSlice, aws.String(key))
+	}
+	return dataSlice
+}
+
+func (ss *StringSet) Set(items []string) {
+	ss.Lock()
+	defer ss.Unlock()
+	for _, item := range items {
+		ss.data.Insert(item)
+	}
+}
+
+func (ss *StringSet) IsEmpty() bool {
+	if ss.data.Len() == 0 {
+		ss.data = sets.NewString()
+		return true
+	}
+	return false
+}
+
+func (ss *StringSet) compareAddDel (items []string) {
```

Comment:
This method should be called `Difference` and simply return a thread-safe result of the underlying `ss.data.Difference()` method:

```go
func (ss *StringSet) Difference(other *StringSet) *StringSet {
    ss.Lock()
    defer ss.Unlock()
    return &StringSet{data: ss.data.Difference(other.data)}
}
```